UCSD/campus-mobile
# Custom App bar not consistently updating when routing to different views#1306

## Summary

This PR is a hotfix for certain cases where the custom app bar title does not change to the corresponding title after navigating. App bar titles should change appropriately in each case. 

## Changelog

[General] [Add] - Added title change methods to cases where the navigation was called and the title didn't change.


## Test Plan

Ways to test the error: **These should not happen anymore with this fix.** 

Steps to reproduce the bug:
1. Go into a dining hall menu through the dining card
2. Press back to go to the home page, and go to a different view (events detail)
3. It would show "Dining" as the title, instead of "Events"

Another test:
1. Deeplinking from campus info card to maps tab should show the app bar title as "UC San Diego", not "Maps".

Alternative way with foreground push notifications:
1. Be in any view other than the notifications tab
2. Send push notification while app is in the foreground and click on the notification
3. The app bar would show the title of whatever the previous screen was, instead of "Notifications"

This is done when:

- [ ]  Custom App bar titles change with the corresponding view
